src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -2382,16 +2382,38 @@ private bool HasGetChildItemDynamicParameters(ProviderInfo providerInfo)
                         // Since the path contained glob characters or we are recursing and the
```

Comment:
@iSazonov, your last commit had 17 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name
```powershell
Expected 8, but got 13.
at <ScriptBlock>, /Users/runner/runners/2.169.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 246
246:             $result.Count | Should -Be 8
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name -Hidden
```powershell
Expected 4, but got 0.
at <ScriptBlock>, /Users/runner/runners/2.169.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 254
254:             $result.Count | Should -Be 4
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name -Directory
```powershell
Expected 2, but got 3.
at <ScriptBlock>, /Users/runner/runners/2.169.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 274
274:             $result.Count | Should -Be 2
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name -Attributes Directory
```powershell
Expected 2, but got 3.
at <ScriptBlock>, /Users/runner/runners/2.169.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 282
282:             $result.Count | Should -Be 2
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name -File
```powershell
Expected 6, but got 10.
at <ScriptBlock>, /Users/runner/runners/2.169.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 290
290:             $result.Count | Should -Be 6
```